### PR TITLE
Add real date math for holidays, fix bug + typos

### PR DESCRIPTION
**Pull request by designatednerd**
In this PR: 
- Used `DateComponents` instead of formatted strings to get the correct date components for holidays
- Added tests for holidays loaded from a calendar, weekends, and weekday good time/bad time
- Fixed a flip-flop of what was good and what was bad
- Fixed a typo

### CODE REVIEW

File: PluginBinary/PluginBinary.swift [hunk 1]
```diff
@@ -25,7 +25,7 @@ extension String: Error { }
             log("Holidays: \(calPath.replacingOccurrences(of: invocation.packagePath + "/", with: ""))")
         }
 
-        let holidayColendars = try invocation.calendarPaths
```
Comment (icanzilb) [inline PluginBinary/PluginBinary.swift:28]: 🙈😂

File: Tests/DoNilDisturbPluginTests/DoNotDisturbPluginTests.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import Foundation
+import iCalendar
+
+@testable import PluginBinary
+
+final class DoNotDisturbPluginTests: XCTestCase {
+    
+    func testHolidayLoad() throws {
+        let date = Foundation.Calendar.current.date(from: DateComponents(year: 2022, month: 8, day: 30, hour: 13))!
+        let path = #filePath
```
Comment (icanzilb) [inline Tests/DoNilDisturbPluginTests/DoNotDisturbPluginTests.swift]: Manually constructing paths isn't safe in tests. Could you please use the API for loading bundle resources? https://developer.apple.com/documentation/xcode/bundling-resources-with-a-swift-package

**Review by icanzilb — CHANGES_REQUESTED** [review]
Hey Ellen, thanks for opening this PR! 

I have only one concern — constructing the resource file path as a string (I left a comment); lmk if you want to adopt the change I suggested or alternatively I can merge and change that file to be an SPM resource myself. 

Either way, much appreciated stopping by to improve that package

File: Tests/DoNilDisturbPluginTests/DoNotDisturbPluginTests.swift (hunk 2 above)
Comment (designatednerd) [reply Tests/DoNilDisturbPluginTests/DoNotDisturbPluginTests.swift]: yep can do

Comment (icanzilb) [reply Tests/DoNilDisturbPluginTests/DoNotDisturbPluginTests.swift]: Much appreciated

**Pull request merged**